ultralytics/engine/exporter.py

```diff
@@ -452,19 +452,27 @@ def serialize(ov_model, file):
 
             def transform_fn(data_item):
                 """Quantization transform function."""
-                assert (
-                    data_item["img"].dtype == torch.uint8
-                ), "Input image must be uint8 for the quantization preprocessing"
-                im = data_item["img"].numpy().astype(np.float32) / 255.0  # uint8 to fp16/32 and 0 - 255 to 0.0 - 1.0
+                assert data_item.dtype == torch.uint8, "Input image must be uint8 for the quantization preprocessing"
+                im = data_item.numpy().astype(np.float32) / 255.0  # uint8 to fp16/32 and 0 - 255 to 0.0 - 1.0
                 return np.expand_dims(im, 0) if im.ndim == 3 else im
 
             # Generate calibration data for integer quantization
             LOGGER.info(f"{prefix} collecting INT8 calibration images from 'data={self.args.data}'")
-            data = check_det_dataset(self.args.data)
-            dataset = YOLODataset(data["val"], data=data, task=self.model.task, imgsz=self.imgsz[0], augment=False)
+            data = (
+                check_det_dataset(self.args.data) if self.args.task != "classify" else check_cls_dataset(self.args.data)
+            )
+            dataset = YOLODataset(
+                data[self.args.split or "val"], data=data, task=self.model.task, imgsz=self.imgsz[0], augment=False
+            )
             n = len(dataset)
             if n < 300:
                 LOGGER.warning(f"{prefix} WARNING ⚠️ >300 images recommended for INT8 calibration, found {n} images.")
+
+            # Batch for quantization
```

Comment:
The dataset itself should be returning the correct batch size, if it's not it likely is simply missing a batch argument and then there is no need for this extra code.

Reply:
I tried adding `batch_size` alone to `YOLODataset` and that didn't seem to work. I had to instead use `build_dataloader` after creating the dataset to get a batched output, so I'm going to update to use this instead.